geonode/harvesting/harvesters/base.py

```diff
@@ -127,20 +156,56 @@ def update_harvesting_session(
 
     def update_geonode_resource(
             self,
-            resource_descriptor: resourcedescriptor.RecordDescription,
+            harvestable_resource_id: int,
             harvesting_session_id: typing.Optional[int] = None
     ):
         """Update GeoNode with the input resource descriptor
 
         This may entail creating the resource if it does not exist yet.
 
         """
-
         # we need the harvester instance in order to know the following information
         # about the resources being created/updated:
         # - Default owner
         # - Default access permissions
         harvester = models.Harvester.objects.get(pk=self.harvester_id)
+        harvestable_resource = models.HarvestableResource.objects.get(
+            pk=harvestable_resource_id)
+        resource_type = self.get_resource_type_class(
+            harvestable_resource.remote_resource_type)
+
+        get_resource_functions = {
+            Map: self.get_map_resource,
+            Layer: self.get_layer_resource,
+            Document: self.get_document_resource,
+        }
+        try:
+            descriptor = get_resource_functions[resource_type](
+                harvestable_resource.unique_identifier)
+            self.update_harvesting_session(
+                harvesting_session_id, additional_harvested_records=1)
+
+            if not descriptor:
+                return
+            defaults = descriptor.to_json()
+            defaults['owner'] = harvester.default_owner
+            uuid = str(defaults['uuid'])
+            manager = ResourceManager()
+            try:
+                _resource = resource_type.objects.get(
+                    uuid=uuid)
+                manager.update(
+                    uuid,
+                    instance=_resource
+                )
+            except resource_type.DoesNotExist:
+                manager.create(
+                    uuid,
+                    resource_type=resource_type,
+                    defaults=defaults
+                )
+        except KeyError:
+            pass
```

Comment:
As said before, this method should not be generating a resource descriptor. It is supposed to have been generated previously and to have been passed as an input parameter to this method.

Reply:
Updated @ricardogsilva